frontend/src/components/Badge/Badge.tsx

```diff
@@ -19,8 +23,21 @@ const Container = styled.div`
   padding: 4px 8px;
   border-radius: 8px;
 
-  background-color: ${({ theme }) => theme.colors.dividers};
-
-  color: ${({ theme }) => theme.colors.textPrimary};
   font: ${({ theme }) => theme.fonts.caption};
+
+  ${({ variant, theme }) => variantStyles[variant](theme)}
 `;
+
+const variantStyles = {
+  default: (theme: Theme) => ({
+    border: 'none',
+    backgroundColor: theme.colors.dividers,
+    color: theme.colors.textPrimary,
+  }),
+  outlinePrimary: (theme: Theme) => ({
+    border: `2px solid ${theme.colors.white}`,
+    backgroundColor: theme.colors.primary,
+    color: theme.colors.white,
+    fontWeight: 600,
+  }),
+};
```

Comment:
만일 variant 별로 객체로 묶지 않는다면, 이렇게 될 것 같습니다!

```suggestion
  border: ${({ variant, theme }) =>
    variant === 'default' ? 'none' : `1px solid ${theme.colors.white}`};

  background-color: ${({ variant, theme }) =>
    variant === 'default' ? theme.colors.dividers : theme.colors.primary};

  color: ${({ variant, theme }) =>
    variant === 'default' ? theme.colors.textPrimary : theme.colors.white};
  font: ${({ theme }) => theme.fonts.caption};
```

Reply:
너도 이 부분에 대해서는 현재 방식이 더 좋은 것 같아요 👍